Automattic/wp-calypso
# Redirect Studio app users to magic login flow when signing up

Related to 5531-gh-Automattic/dotcom-forge, 5532-gh-Automattic/dotcom-forge 

## Proposed Changes

* With this PR, users who sign up for WordPress.com via the Studio app will be redirected to the magic login form. 
* This change will prevent a poor experience, in which users who signed up following the Studio flow were not correctly logged into their WordPress.com account. It will also avoid users being directed through the site creation flow. Further context for this decision can be found [here](5531-gh-Automattic/dotcom-forge#issuecomment-2084811812).

| Signup form  | Confirmation page |
| ------------- | ------------- |
| <img src="https://github.com/Automattic/wp-calypso/assets/2998162/49ed91f9-7f74-4499-987e-140e4ec23ae5" width="100%"> | <img src="https://github.com/Automattic/wp-calypso/assets/2998162/11e5bda2-783e-400b-9ad4-aae1e0ee73e2" width="100%"> |

>  [!NOTE]
> There is a `Create free account` link in the Studio app that will be updated after this PR has been merged.

## Testing Instructions

* Log out of WordPress.com and navigate to [the Calypso Live link in this comment](https://github.com/Automattic/wp-calypso/pull/90204#issuecomment-2090154172).
* Add the following to the end of the Calypso Live link in order to simular the URL that's clicked from the Studio app: `/log-in?client_id=95109&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D95109%26redirect_uri%3Dwpcom-local-dev%253A%252F%252Fauth%26scope%3Dglobal%26from-calypso%3D1`
* Select the option to create a new account from the login page.
* Submit an email address that's not already associated with a WordPress.com account.
* Click the verification link in the new email and verify that you're taken to authentication page for the Studio app.
* Confirm the following tracks events have been logged:
   * `calypso_login_block_login_form_send_account_create_magic_link` with a `client_id: 95109` property
   * `calypso_login_email_link_success` with a `client_id: 95109` property 

## Pre-merge Checklist
